src/GovUk.Education.ExploreEducationStatistics.Admin/Migrations/ContentMigrations/20210223130444_EES1710AddMethodologyFile.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Migrations.ContentMigrations
+{
+    public partial class EES1710AddMethodologyFile : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.CreateTable(
+                name: "MethodologyFiles",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(nullable: false),
+                    MethodologyId = table.Column<Guid>(nullable: false),
+                    FileId = table.Column<Guid>(nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_MethodologyFiles", x => x.Id);
+                    table.ForeignKey(
+                        name: "FK_MethodologyFiles_Files_FileId",
+                        column: x => x.FileId,
+                        principalTable: "Files",
+                        principalColumn: "Id",
+                        onDelete: ReferentialAction.Cascade);
```

Comment:
It actually means if the File is deleted, delete all the MethodologyFile links for that file.
I did it this way to be the same as ReleaseFile where the same is true, and you can see that with this test on the local database:

`
-- Observe that the releasefile links for the file exist. There should be 1 result.
select * from ReleaseFiles where FileId = '15C05193-1B4A-4ACB-6A81-08D7EC2D8168'
-- Delete dataimport for File to allow deleting the file.
delete from DataImports where FileId = '15C05193-1B4A-4ACB-6A81-08D7EC2D8168'
-- Delete the file, in turn cascade deletes the releasefile links.
delete from files where Id = '15C05193-1B4A-4ACB-6A81-08D7EC2D8168'
-- Observe that there are no releasefile links.
select * from ReleaseFiles where FileId = '15C05193-1B4A-4ACB-6A81-08D7EC2D8168'
`

I think I will change these to not have this cascade delete, so that all links have to be removed first. (Where we delete Release files we already unlink them first anyway).

By the way, in terms of the other relationship between MethodologyFile and Methodology, that's already set to take no action, i.e. don't delete any MethodologyFiles when deleting the Methodology. I think we will write versioning in the same way as did for Releases, so when deleting a Methodology version we will unlink the files from the Methodology first, check if they are linked to any other Methodologies and make sure they are deleted from storage if no longer used.

Reply:
I've made this change now.